File: 1758-minimum-changes-to-make-alternating-binary-string/1758. Minimum Changes To Make Alternating Binary String.py
class Solution:
    def minOperations(self, s: str) -> int: 
        a,b = 0, 0
        # 010101
        zero_turn = True
        for c in s:
            if int(c) != zero_turn:
                a += 1
            zero_turn = not zero_turn
        
        zero_turn = False
        for c in s:
            if int(c) != zero_turn:
                b += 1
            zero_turn = not zero_turn


        return min(a,b)

File: 1758-minimum-changes-to-make-alternating-binary-string/1758. Minimum Changes To Make Alternating Binary String.cpp
class Solution {
public:
    int minOperations(string s) {
        // start 0
        int a = 0;
        for (int i = 0; i < s.size(); i++) {
            if (i%2 == 0) {
                a += s[i] != '0';
            } else {
                a += s[i] != '1';
            }
        }
        // start 1
        int b = 0;
        for (int i = 0; i < s.size(); i++) {
            if (i%2 == 0) {
                b += s[i] != '1';
            } else {
                b += s[i] != '0';
            }
        }

        return min(a,b);
    }
};